tests/unit/igraph_circulant.c

```diff
@@ -0,0 +1,195 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+
+#include "test_utilities.inc"
+
+int main() {
+
+    igraph_t graph, graph_test;
+    igraph_vector_int_t l;
+    igraph_bool_t iso;
+    igraph_integer_t i;
+
+    /* Testing invalid values */
+
+    /* n = -3, l = [1], undirected */
+    igraph_vector_int_init(&l, 0);
+    igraph_vector_int_push_back(&l, 1);
+    CHECK_ERROR(igraph_circulant(&graph, -3, &l, IGRAPH_UNDIRECTED), IGRAPH_EINVAL);
+    igraph_vector_int_destroy(&l);
+
+    /* Testing n = 0 case */
+    /* n = 0, l = [1], undirected */
+
+    igraph_vector_int_init(&l, 0);
+    igraph_vector_int_push_back(&l, 1);
+    IGRAPH_ASSERT(igraph_circulant(&graph, 0, &l, IGRAPH_UNDIRECTED) == IGRAPH_SUCCESS);
+    IGRAPH_ASSERT(igraph_vcount(&graph) == 0);
+    IGRAPH_ASSERT(igraph_ecount(&graph) == 0);
+    igraph_vector_int_destroy(&l);
+
+
+    /* Testing n = 1 case */
+    /* n = 1, l = [1], undirected */
+    igraph_vector_int_init(&l, 0);
+    igraph_vector_int_push_back(&l, 1);
+    IGRAPH_ASSERT(igraph_circulant(&graph, 1, &l, IGRAPH_UNDIRECTED) == IGRAPH_SUCCESS);
+    igraph_small(&graph_test, 1, IGRAPH_UNDIRECTED, -1);
+    IGRAPH_ASSERT(igraph_is_same_graph(&graph, &graph_test, &iso) == IGRAPH_SUCCESS);
+    IGRAPH_ASSERT(iso);
+    igraph_destroy(&graph);
+    igraph_destroy(&graph_test);
+    igraph_vector_int_destroy(&l);
+
+    /* Testing n = 2 case */
+    /* n = 2, l = [1], undirected */
+    igraph_vector_int_init(&l, 0);
+    igraph_vector_int_push_back(&l, 1);
+    IGRAPH_ASSERT(igraph_circulant(&graph, 2, &l, IGRAPH_UNDIRECTED) == IGRAPH_SUCCESS);
+    igraph_small(&graph_test, 2, IGRAPH_UNDIRECTED, 0, 1, -1);
+    IGRAPH_ASSERT(igraph_is_same_graph(&graph, &graph_test, &iso) == IGRAPH_SUCCESS);
```

Comment:
Nitpick: In several places, you use the name `iso` with `igraph_is_same_graph`. Can you please change this to a different name, such as `same`? `igraph_is_same_graph` does not test isomorphism, but checks if the graphs have the same edges and vertices. While you understand that, I would like to point other beginners to your PR to show them how a contribution goes. I worry they may be confused by this.

Reply:
Done